### DISCUSSION TITLE
Autosquash in interactive rebase

### DISCUSSION THREAD

**Discussion by folago** [jesseduffield/lazygit#1160] [Q&A] [answered]
Is there a way to get the fixup and squash commit already sorted during an interactive rebase?
Basically the same as `git rebase -i --autosquash`.
When rebasing with a lot of fixup after a review is quite convenient to have them all sorted out and not have to do it by hand.

edit: typo and clarification

**Comment by jesseduffield** [comment, marked as answer]
First off: apologies for the late reply, github didn't send me notifications about comments in these discussion threads!

If I understand your question correctly, you might find value in the shift+S keybinding to squash all `fixup!` commits above the selected commit: 
![image](https://user-images.githubusercontent.com/8456633/114302191-45a02200-9b0b-11eb-923a-e5a9b0426f3e.png)

Does that satisfy your use case?

**Reply by folago** [reply]
> First off: apologies for the late reply, github didn't send me notifications about comments in these discussion threads!
> 

No problem :)

> If I understand your question correctly, you might find value in the shift+S keybinding to squash all `fixup!` commits above the selected commit:
> ![image](https://user-images.githubusercontent.com/8456633/114302191-45a02200-9b0b-11eb-923a-e5a9b0426f3e.png)
> 
> Does that satisfy your use case?

A-HA yesss!
This is perfect. I did not notice it, thanks for pointing it out.
And thanks for lazygit!

**Reply by jesseduffield** [reply]
no worries :)

**Reply by folago** [reply]
This got me thinking that maybe I saw that option before but I thought it would squash all the commits above into one, not just the fixups. Maybe calling it autosquash would be more intuitive. Or mention autosquash/fixups in the command description.
My two cents :)

**Reply by jesseduffield** [reply]
Good call. I've updated it to `squash all 'fixup!' commits above selected commit (autosquash)` here: https://github.com/jesseduffield/lazygit/pull/1260

**Comment by joebb97** [comment]
Hi there,

First off I love lazygit and think it is great. Thanks for your work on it.

I don't think shift+S is the same as what was originally asked.

Seems like shift+S is for use outside of an interactive rebase, since that shortcut is not enabled while rebasing.

An option I would like to see is to start an interactive rebase and have all the fixup commits be moved to right above the commit they fixup, and then marked with fixup, just like you would get doing an interactive rebase with `--autosquash` in plain old git CLI.

Pressing `i` and then having all my fixup commits be marked as `pick` and having to move them manually to where they should go is not ideal.

I hope you'll consider such a feature.

edit: I suppose a workaround is to run `shift+S` before you start the interactive rebase, but that feels unintuitive.

**Reply by stefanhaller** [reply]
> edit: I suppose a workaround is to run `shift+S` before you start the interactive rebase, but that feels unintuitive.

This is exactly what I do, and I think it's perfect. What's wrong with it? When I want to do any further rebasing beyond squashing fixups, I don't want to have to deal with all the noise from the unsquashed fixup commits, so I first press `shift-S` and then `i`.
